Sources/xcprojectlint-package/NoDanglingTests.swift

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2018 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Foundation
+
+public func noDanglingTests(_ project: Project, errorReporter: ErrorReporter) -> Int32 {
+    var result = EX_DATAERR
+
+    let targetFiles = Set(project.buildFiles.map{$1.fileRef})
+    let excludedExtensions = Set(["xctest", "h", "xcconfig", "xcfilelist"])
+    let actualFilesInProjectNavigator = project.fileReferences
+        .filter{ key, value in
+            let fileComponents = value.title.components(separatedBy: ".")
+            guard fileComponents.count > 1,
+                let fileName = fileComponents.first?.hasSuffix("Tests"),
```

Comment:
I suggest that we take a more generalised approach to this problem.
Instead of looking for dangling files with `Tests` suffix and managing `excludedExtensions` black list, we can look for all source files that are missing target memberships and manage white list.

For example, we could say that all `[.swift, .m]` files, that are not on`project.buildFiles` list should be reported. 
This will cover all dangling `*Tests` files as well as other source files and be agnostic to project specific naming conventions like `*UnitTest` and conflicting file names like `HowToWriteTests.md`, that would otherwise have to be added to black list by `xcprojectlist` users.

Reply:
I think this is a great idea. I wish I'd had it. :-)

Reply:
well, this will hurt, because storyboards, info.plists etc all are not under a target membership.

Reply:
I'll try with `let extensionsWhiteList = Set(["m", "swift"])` but I think it would be better to contain c++ files like `.mm` as well.

Reply:
Sure, `.swift, .m, .cpp, .mm, .c` should all be there. 
I just outlined the idea for white list, don't take it as an exhaustive list :)